### PR TITLE
Run hlint in CI again.

### LINKED ISSUE #137: Add script/lint and use it in CI.
We have a .hlint.yaml that we are not using.

### PR DESCRIPTION
**Pull request by patrickt**
- [x] Depends on #352 

Fixes #137

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -46,6 +46,8 @@ script:
 - cabal v2-run semantic-source:doctest
 # parse-examples is disabled because it slaughters our CI
 # - cabal v2-run semantic:parse-examples
+# Downloads an hlint binary rather than going through cabal-install
+- curl -sSL https://raw.github.com/ndmitchell/hlint/master/misc/run.sh | sh -s src semantic-python
```
Comment (robrix) [inline .travis.yml]: Can we download the thing as part of the `install` step instead?

File: .hlint.yaml [hunk 2]
```diff
@@ -45,8 +49,29 @@
 
 - ignore: {name: Use ., within: [Analysis.Abstract.Graph.graphingModules, Semantic.Distribute]}
 
-- ignore: {name: Reduce duplication, within: [Semantic.Util, Semantic.UtilDisabled]}
-- ignore: {name: Use newtype instead of data, within: [Semantic.Api.V1.CodeAnalysisPB]}
+- ignore:
+    within:
+      - Proto.Semantic
+      - Proto.Semantic_Fields
+      - Proto.Semantic_JSON
```
Comment (robrix) [inline .hlint.yaml]: Good call 👍

File: src/Data/ImportPath.hs [hunk 3]
```diff
@@ -19,7 +19,7 @@ data ImportPath = ImportPath { unPath :: FilePath, pathIsRelative :: IsRelative
 importPath :: Text -> ImportPath
 importPath str = let path = stripQuotes str in ImportPath (T.unpack path) (pathType path)
   where
-    pathType xs | not (T.null xs), T.head xs == '.' = Relative -- head call here is safe
+    pathType xs | not (T.null xs), fmap fst (T.uncons xs) == Just '.' = Relative -- head call here is safe
```
Comment (robrix) [inline src/Data/ImportPath.hs]: I don’t think this is an improvement. That said, if we decide to keep it, we should 🔥 the `T.null` guard immediately before it.

Comment (patrickt) [reply src/Data/ImportPath.hs]: I’ll refactor this a bit.

Comment (patrickt) [reply src/Data/ImportPath.hs]: This is a bit more readable now, eliding the `null` and `head` in favor of an `uncons` brought out into its own def.

File: .travis.yml [hunk 4]
```diff
@@ -24,16 +24,22 @@ matrix:
...
 - cabal --version
+- hlint --version
 
 install:
 - cabal v2-update -v
 - cabal v2-configure --enable-tests --enable-benchmarks --disable-optimization --write-ghc-environment-files=always --jobs=2
 - cabal v2-build --only-dependencies
 
 script:
+- hlint src semantic-python
```
Comment (robrix) [inline .travis.yml]: 🙇 — let’s extend this to the other packages too in future PRs

**Pull request merged**